packages/core/framework/src/http/middlewares/ensure-publishable-api-key.ts

```diff
@@ -0,0 +1,77 @@
+import { Query } from "@medusajs/types"
+import {
+  ApiKeyType,
+  ContainerRegistrationKeys,
+  isPresent,
+  MedusaError,
+  PUBLISHABLE_KEY_HEADER,
+} from "@medusajs/utils"
+import {
+  MedusaNextFunction,
+  MedusaResponse,
+  MedusaStoreRequest,
+} from "../../http"
+
+export async function ensurePublishableApiKeyMiddleware(
+  req: MedusaStoreRequest,
+  _res: MedusaResponse,
+  next: MedusaNextFunction
+) {
+  const publishableApiKey = req.get(PUBLISHABLE_KEY_HEADER)
+
+  if (!isPresent(publishableApiKey)) {
+    try {
+      throw new MedusaError(
+        MedusaError.Types.NOT_ALLOWED,
+        `Publishable API key required in the request header: ${PUBLISHABLE_KEY_HEADER}. You can manage your keys in settings in the dashboard.`
+      )
+    } catch (e) {
+      return next(e)
+    }
```

Comment:
Q: Do you remember why we are not doing 😂 
```ts
const error = new MedusaError(...)
return next(error)
```

Reply:
I tried this and the middleware got stuck 🤷🏻